nexus/db-model/src/deployment.rs

```diff
@@ -0,0 +1,263 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Types for representing the deployed software and configuration in the
+//! database
+
+use crate::inventory::ZoneType;
+use crate::omicron_zone_config::{OmicronZone, OmicronZoneNic};
+use crate::schema::{
+    blueprint, bp_omicron_zone, bp_omicron_zone_nic,
+    bp_omicron_zones_not_in_service, bp_sled_omicron_zones, bp_target,
+};
+use crate::{ipv6, Generation, MacAddr, Name, SqlU16, SqlU32, SqlU8};
+use chrono::{DateTime, Utc};
+use ipnetwork::IpNetwork;
+use nexus_types::deployment::BlueprintTarget;
+use uuid::Uuid;
+
+/// See [`nexus_types::deployment::Blueprint`].
+#[derive(Queryable, Insertable, Clone, Debug, Selectable)]
+#[diesel(table_name = blueprint)]
+pub struct Blueprint {
+    pub id: Uuid,
+    pub parent_blueprint_id: Option<Uuid>,
+    pub time_created: DateTime<Utc>,
+    pub creator: String,
+    pub comment: String,
+}
+
+impl From<&'_ nexus_types::deployment::Blueprint> for Blueprint {
+    fn from(bp: &'_ nexus_types::deployment::Blueprint) -> Self {
+        Self {
+            id: bp.id,
+            parent_blueprint_id: bp.parent_blueprint_id,
+            time_created: bp.time_created,
+            creator: bp.creator.clone(),
+            comment: bp.comment.clone(),
+        }
+    }
+}
+
+impl From<Blueprint> for nexus_types::deployment::BlueprintMetadata {
+    fn from(value: Blueprint) -> Self {
+        Self {
+            id: value.id,
+            parent_blueprint_id: value.parent_blueprint_id,
+            time_created: value.time_created,
+            creator: value.creator,
+            comment: value.comment,
+        }
+    }
+}
+
+/// See [`nexus_types::deployment::BlueprintTarget`].
+#[derive(Queryable, Clone, Debug, Selectable, Insertable)]
+#[diesel(table_name = bp_target)]
+pub struct BpTarget {
+    pub version: i64, // i64 only for db serialization; should never be negative
```

Comment:
Sure, switched to `SqlU32`. I wasn't 100% sure that was big enough to never worry about exhausting it, but I think I've convinced myself it's fine. Even if runaway automation set a new target every second (which seems pretty unlikely, since that first requires generating and inserting a new blueprint in its entirety), u32 is fine for 100+ years.